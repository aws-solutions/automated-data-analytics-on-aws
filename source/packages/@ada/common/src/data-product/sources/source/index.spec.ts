/*! Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import { SouceDetailsSchema } from './index';

describe('common/data-product/source', () => {
  describe('SouceDetailsSchema', () => {
    it('should correctly merge all source details schemas as partials', () => {
      expect(SouceDetailsSchema).toMatchSnapshot();
    });
  });
});
