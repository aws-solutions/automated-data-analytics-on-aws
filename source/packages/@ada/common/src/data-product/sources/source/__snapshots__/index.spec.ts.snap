// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/data-product/source SouceDetailsSchema should correctly merge all source details schemas as partials 1`] = `
Object {
  "definitions": Object {
    "GSLocation": Object {
      "id": "GSLocation",
      "properties": Object {
        "bucket": Object {
          "maxLength": 63,
          "minLength": 3,
          "pattern": "^[a-z0-9]+[a-z0-9-]+[a-z0-9]+$",
          "type": "string",
        },
        "key": Object {
          "maxLength": 1024,
          "minLength": 1,
          "pattern": "^[a-zA-Z0-9!_.*'()-]+(/[a-zA-Z0-9!_.*'()-]+)*$",
          "type": "string",
        },
      },
      "required": Array [
        "bucket",
        "key",
      ],
      "schema": "http://json-schema.org/draft-04/schema#",
      "type": "object",
    },
    "GoogleAnalyticsIdentifier": Object {
      "id": "GoogleAnalyticsIdentifier",
      "properties": Object {
        "dimensions": Object {
          "maxLength": 20480,
          "minLength": 4,
          "pattern": "^(ga:[a-zA-Z0-9]+,?)+$",
          "type": "string",
        },
        "metrics": Object {
          "maxLength": 20480,
          "minLength": 4,
          "pattern": "^(ga:[a-zA-Z0-9]+,?)+$",
          "type": "string",
        },
        "scheduleRate": Object {
          "type": "string",
        },
        "since": Object {
          "maxLength": 64,
          "pattern": "^(\\\\d{4})-(\\\\d{2})-(\\\\d{2})(?:T(\\\\d{2}):(\\\\d{2}):(\\\\d{2}(?:\\\\.\\\\d*)?)((-(\\\\d{2}):(\\\\d{2})|Z)?))?$",
          "type": "string",
        },
        "until": Object {
          "maxLength": 64,
          "pattern": "^(\\\\d{4})-(\\\\d{2})-(\\\\d{2})(?:T(\\\\d{2}):(\\\\d{2}):(\\\\d{2}(?:\\\\.\\\\d*)?)((-(\\\\d{2}):(\\\\d{2})|Z)?))?$",
          "type": "string",
        },
        "viewId": Object {
          "type": "string",
        },
      },
      "required": Array [
        "viewId",
        "dimensions",
        "metrics",
      ],
      "schema": "http://json-schema.org/draft-04/schema#",
      "type": "object",
    },
    "GoogleBigQueryTableIdentifier": Object {
      "id": "GoogleBigQueryTableIdentifier",
      "properties": Object {
        "query": Object {
          "type": "string",
        },
      },
      "required": Array [
        "query",
      ],
      "schema": "http://json-schema.org/draft-04/schema#",
      "type": "object",
    },
    "GoogleServiceAccountAuth": Object {
      "id": "GoogleServiceAccountAuth",
      "properties": Object {
        "clientEmail": Object {
          "description": "The email address of the service account",
          "pattern": "^[^@]+@[^@]+\\\\.iam\\\\.gserviceaccount\\\\.com$",
          "title": "Client Email",
          "type": "string",
        },
        "clientId": Object {
          "description": "The client id of the Google Service Account",
          "title": "Client ID",
          "type": "string",
        },
        "privateKey": Object {
          "description": "The private key of the service account",
          "pattern": "^(-{5}BEGIN PRIVATE KEY-{5}\\\\s(([\\\\w+/=]{63,76}\\\\s){10,30})([\\\\w+/=]{1,76}\\\\s)?-{5}END PRIVATE KEY-{5})\\\\s?$",
          "title": "Private Key",
          "type": "string",
        },
        "privateKeyId": Object {
          "description": "The private key id of the Google Service Account",
          "title": "Private Key ID",
          "type": "string",
        },
        "privateKeySecretName": Object {
          "type": "string",
        },
        "projectId": Object {
          "description": "The ID of the Google Cloud Project",
          "maxLength": 30,
          "minLength": 6,
          "pattern": "^[a-z][a-z0-9-]{4,28}[a-z0-9]$",
          "title": "Project Id",
          "type": "string",
        },
      },
      "required": Array [
        "projectId",
        "clientEmail",
        "clientId",
        "privateKeyId",
      ],
      "schema": "http://json-schema.org/draft-04/schema#",
      "type": "object",
    },
    "S3Location": Object {
      "id": "S3Location",
      "properties": Object {
        "bucket": Object {
          "maxLength": 63,
          "minLength": 3,
          "pattern": "^[a-z0-9]+[a-z0-9-]+[a-z0-9]+$",
          "type": "string",
        },
        "key": Object {
          "maxLength": 1024,
          "minLength": 1,
          "pattern": "^[a-zA-Z0-9!_.*'()-]+(/[a-zA-Z0-9!_.*'()-]+)*$",
          "type": "string",
        },
      },
      "required": Array [
        "bucket",
        "key",
      ],
      "schema": "http://json-schema.org/draft-04/schema#",
      "type": "object",
    },
    "SourceDetailsFileUpload": Object {
      "id": "SourceDetailsFileUpload",
      "properties": Object {
        "bucket": Object {
          "maxLength": 63,
          "minLength": 3,
          "pattern": "^[a-z0-9]+[a-z0-9-]+[a-z0-9]+$",
          "type": "string",
        },
        "key": Object {
          "maxLength": 1024,
          "minLength": 1,
          "pattern": "^[a-zA-Z0-9!_.*'()-]+(/[a-zA-Z0-9!_.*'()-]+)*$",
          "type": "string",
        },
      },
      "required": Array [
        "bucket",
        "key",
      ],
      "schema": "http://json-schema.org/draft-04/schema#",
      "type": "object",
    },
    "SourceDetailsGoogleAnalytics": Object {
      "allOf": Array [
        Object {
          "ref": "#/definitions/GoogleAnalyticsIdentifier",
        },
        Object {
          "ref": "#/definitions/GoogleServiceAccountAuth",
        },
      ],
      "id": "SourceDetailsGoogleAnalytics",
      "schema": "http://json-schema.org/draft-04/schema#",
      "type": "object",
    },
    "SourceDetailsGoogleBigQuery": Object {
      "allOf": Array [
        Object {
          "ref": "#/definitions/GoogleBigQueryTableIdentifier",
        },
        Object {
          "ref": "#/definitions/GoogleServiceAccountAuth",
        },
      ],
      "id": "SourceDetailsGoogleBigQuery",
      "schema": "http://json-schema.org/draft-04/schema#",
      "type": "object",
    },
    "SourceDetailsGoogleStorage": Object {
      "allOf": Array [
        Object {
          "ref": "#/definitions/GSLocation",
        },
        Object {
          "ref": "#/definitions/GoogleServiceAccountAuth",
        },
      ],
      "id": "SourceDetailsGoogleStorage",
      "schema": "http://json-schema.org/draft-04/schema#",
      "type": "object",
    },
    "SourceDetailsKinesis": Object {
      "id": "SourceDetailsKinesis",
      "properties": Object {
        "kinesisStreamArn": Object {
          "maxLength": 2048,
          "pattern": "^arn:[\\\\w]+:[\\\\w-]+:[\\\\w-]+:[\\\\d]+:[*-\\\\/\\\\w]+$",
          "type": "string",
        },
      },
      "required": Array [
        "kinesisStreamArn",
      ],
      "schema": "http://json-schema.org/draft-04/schema#",
      "type": "object",
    },
    "SourceDetailsS3": Object {
      "id": "SourceDetailsS3",
      "properties": Object {
        "bucket": Object {
          "maxLength": 63,
          "minLength": 3,
          "pattern": "^[a-z0-9]+[a-z0-9-]+[a-z0-9]+$",
          "type": "string",
        },
        "key": Object {
          "maxLength": 1024,
          "minLength": 1,
          "pattern": "^[a-zA-Z0-9!_.*'()-]+(/[a-zA-Z0-9!_.*'()-]+)*$",
          "type": "string",
        },
      },
      "required": Array [
        "bucket",
        "key",
      ],
      "schema": "http://json-schema.org/draft-04/schema#",
      "type": "object",
    },
  },
  "id": "SouceDetailsSchema",
  "properties": Object {
    "bucket": Object {
      "maxLength": 63,
      "minLength": 3,
      "pattern": "^[a-z0-9]+[a-z0-9-]+[a-z0-9]+$",
      "type": "string",
    },
    "clientEmail": Object {
      "description": "The email address of the service account",
      "pattern": "^[^@]+@[^@]+\\\\.iam\\\\.gserviceaccount\\\\.com$",
      "title": "Client Email",
      "type": "string",
    },
    "clientId": Object {
      "description": "The client id of the Google Service Account",
      "title": "Client ID",
      "type": "string",
    },
    "dimensions": Object {
      "maxLength": 20480,
      "minLength": 4,
      "pattern": "^(ga:[a-zA-Z0-9]+,?)+$",
      "type": "string",
    },
    "key": Object {
      "maxLength": 1024,
      "minLength": 1,
      "pattern": "^[a-zA-Z0-9!_.*'()-]+(/[a-zA-Z0-9!_.*'()-]+)*$",
      "type": "string",
    },
    "kinesisStreamArn": Object {
      "maxLength": 2048,
      "pattern": "^arn:[\\\\w]+:[\\\\w-]+:[\\\\w-]+:[\\\\d]+:[*-\\\\/\\\\w]+$",
      "type": "string",
    },
    "metrics": Object {
      "maxLength": 20480,
      "minLength": 4,
      "pattern": "^(ga:[a-zA-Z0-9]+,?)+$",
      "type": "string",
    },
    "privateKey": Object {
      "description": "The private key of the service account",
      "pattern": "^(-{5}BEGIN PRIVATE KEY-{5}\\\\s(([\\\\w+/=]{63,76}\\\\s){10,30})([\\\\w+/=]{1,76}\\\\s)?-{5}END PRIVATE KEY-{5})\\\\s?$",
      "title": "Private Key",
      "type": "string",
    },
    "privateKeyId": Object {
      "description": "The private key id of the Google Service Account",
      "title": "Private Key ID",
      "type": "string",
    },
    "privateKeySecretName": Object {
      "type": "string",
    },
    "projectId": Object {
      "description": "The ID of the Google Cloud Project",
      "maxLength": 30,
      "minLength": 6,
      "pattern": "^[a-z][a-z0-9-]{4,28}[a-z0-9]$",
      "title": "Project Id",
      "type": "string",
    },
    "query": Object {
      "type": "string",
    },
    "scheduleRate": Object {
      "type": "string",
    },
    "since": Object {
      "maxLength": 64,
      "pattern": "^(\\\\d{4})-(\\\\d{2})-(\\\\d{2})(?:T(\\\\d{2}):(\\\\d{2}):(\\\\d{2}(?:\\\\.\\\\d*)?)((-(\\\\d{2}):(\\\\d{2})|Z)?))?$",
      "type": "string",
    },
    "until": Object {
      "maxLength": 64,
      "pattern": "^(\\\\d{4})-(\\\\d{2})-(\\\\d{2})(?:T(\\\\d{2}):(\\\\d{2}):(\\\\d{2}(?:\\\\.\\\\d*)?)((-(\\\\d{2}):(\\\\d{2})|Z)?))?$",
      "type": "string",
    },
    "viewId": Object {
      "type": "string",
    },
  },
  "schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
}
`;
